apps/metric_vnode/src/metric_vnode.erl

```diff
@@ -230,25 +230,26 @@ handle_command({get, ReqID, Bucket, Metric, {Time, Count}}, Sender,
             {Resolution, State1} = get_resolution(Bucket, State),
             {reply, {ok, ReqID, Idx, {Resolution, Data}}, State1};
         %% The request is neither before, after nor entirely inside the cache
-        %% we sadly have to flush it.
-        %% This are the conditions where we have to flush the cache and then
-        %% read it from the ui server
+        %% we have to read data, but apply cached part on top of it.
         [{BM, Start, Size, _Time, Array}]
           when
               %% The window starts before the cache but ends in it
               (Time < Start andalso Time + Count >= Start)
+
               %% The window starts inside the cahce and ends afterwards
               orelse (Time =< Start + Size andalso Time + Count > Start + Size)
               ->
-            Bin = k6_bytea:get(Array, 0, Size * ?DATA_SIZE),
-            k6_bytea:delete(Array),
-            ets:delete(T, {Bucket, Metric}),
-            metric_io:write(IO, Bucket, Metric, Start, Bin),
-            metric_io:read(IO, Bucket, Metric, Time, Count, ReqID, Sender),
+            PartStart = max(Time, Start),
+            PartCount = min(Time + Count, Start + Size) - PartStart,
+            SkipBytes = (PartStart - Start),
+            Bin = k6_bytea:get(Array, SkipBytes, (PartCount * ?DATA_SIZE)),
+            Offset = PartStart - Time,
+            Part = {Offset, PartCount, Bin},
+            metric_io:read_rest(IO, Bucket, Metric, Time, Count, Part, ReqID, Sender),
```

Comment:
Line 248 is too long:             metric_io:read_rest(IO, Bucket, Metric, Time, Count, Part, ReqID, Sender),.